data/lists.js

```diff
@@ -3,3 +3,168 @@
 // edit list - used by /list/:id route
 // delete list - used by /list/:id route
 // delete entry from list - used by /list/:id route
+import { users } from "../config/mongoCollections.js";
+import { ObjectId } from "mongodb";
+import * as validation from "../utils/validation.js";
+
+const exportedMethods = {
+  async create(userId, title, movies) {
+    //for validation validate if
+    //userId is valid
+    //title is a string
+    //movies is an array of integers
+    validation.checks(userId);
+    userId = userId.trim();
+    if (!ObjectId.isValid(userId)) throw "invalid object ID";
+    title = title.trim();
+    if (!movies || !Array.isArray(movies))
+      throw "You must provide an array of movies";
+    if (movies.length === 0) throw "You must supply at least one movie";
+    movies.forEach((elem) => {
+      elem = validation.checkNumber(elem, "movie");
+    });
+    const userCollection = await users();
+    const user = await userCollection.findOne({ _id: new ObjectId(userId) });
+    if (user === null) throw "No user with that id";
+
+    let newList = {
+      _id: new ObjectId(),
+      title: title,
+      movies: movies,
+    };
+    const updatedInfo = await userCollection.updateOne(
+      { _id: new ObjectId(userId) },
+      {
+        $push: { lists: newList },
+      }
+    );
+    if (updatedInfo.modifiedCount === 0) {
+      throw "No update occured in the user while adding album!";
+    }
+
+    newList._id = newList._id.toString();
+    return newList;
+  },
+
+  async getAll(userId) {
+    //check userID
+    validation.checks(userId);
+    userId = userId.trim();
+    if (!ObjectId.isValid(userId)) throw "invalid object ID";
+    const userCollection = await users();
+    const user = await userCollection.findOne({ _id: new ObjectId(userId) });
+    if (user === null) throw "No user with that id";
+
+    let movieList = user.lists;
+    if (movieList.length === 0) throw "No lists present for this user Id";
+    movieList.forEach((element) => {
+      element._id = element._id.toString();
+    });
+    return movieList;
+  },
+
+  async get(listId) {
+    validation.checks(listId);
+    listId = listId.trim();
+    if (!ObjectId.isValid(listId)) throw "invalid object ID";
+
+    const userCollection = await users();
+    const list = await userCollection.findOne(
+      { "lists._id": new ObjectId(listId) },
+      {
+        projection: {
+          _id: 0,
+          lists: { $elemMatch: { _id: new ObjectId(listId) } },
+        },
+      }
+    );
+    if (!list) {
+      throw "Error: list not found";
+    }
+    list.lists[0]._id = list.lists[0]._id.toString();
+    return list.lists[0];
+  },
+  async delete(listId) {
+    validation.checks(listId);
+    listId = listId.trim();
+    // console.log("hi3");
+    if (!ObjectId.isValid(listId)) throw "invalid object ID";
+    // console.log("hi3");
```

Comment:
Same as the first comment, you can use the existing validation function for that:
```suggestion
    listId= validation.checkId(listId, "ListID")
```